src/Shared/Application/EventListener/MethodNotAllowedListener.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Shared\Application\EventListener;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+final class MethodNotAllowedListener
+{
+    private const ERROR_TYPE = '/errors/405';
+    private const ERROR_TITLE = 'Method Not Allowed';
+    private const ERROR_DETAIL =
+        'The requested HTTP method is not allowed on this resource.';
+
+    /**
+     * @var array<string, array<int, string>>
+     */
+    private array $restrictions = [
```

Comment:
It will be better to retrieve this information from the api platform and OpenAPI instead of hardcoding this information
Please follow always DRY principle